### ISSUE TITLE
Compatibility with Ruby 2.4.1

### ISSUE THREAD

**Issue by ACPK** [brigade/haml-lint#214]
When using the Atom haml-lint package, I started receiving the following error after updating Ruby:

    linter-registry.js [sm]:144 [Linter] Error running HAML-Lint Error: warning: parser/current is     loading parser/ruby24, which recognizes
    warning: 2.4.0-compliant syntax, but you are running 2.4.1.
    warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.

**Comment by trotzig** [comment]
Does the same error happen if you run `haml-lint` directly from the command line?

**Comment by ACPK** [comment]
@trotzig -Yes, it does. Here's the logs.

    MacBook-Pro-2% haml-lint
    warning: parser/current is loading parser/ruby24, which recognizes
    warning: 2.4.0-compliant syntax, but you are running 2.4.1.
    warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.

**Comment by michaelherold** [comment]
That warning is raised whenever the parser gem is loading a syntax that is different than the current version of MRI. Parser has not been updated for 2.4.1 yet so there's nothing we can do right now. The PR for that is here: whitequark/parser#342

**Comment by b4ugh** [comment]
I see the PR @michaelherold linked was closed - however I am still getting this error when running haml-lint in both atom and through the command line.

`warning: parser/current is loading parser/ruby24, which recognizes
warning: 2.4.0-compliant syntax, but you are running 2.4.1.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.`

**Comment by sds** [comment]
Hey @b4ugh, it's not clear why https://github.com/whitequark/parser/pull/342 was closed. whitequark [mentioned](https://github.com/whitequark/parser/pull/342#issuecomment-289304128) that the fix was blocked by https://github.com/whitequark/parser/issues/336, which hasn't been closed yet.

In other words: the upstream issue does not appear to be fixed yet.

**Comment by twp88** [comment]
Is anyone aware if this issue has been fixed yet?

**Comment by michaelherold** [comment]
No, it appears that whitequark is still trying to catch up on a backlog that built up due to some health issues. The message is merely a warning and does not affect the functionality of the gem.

**Comment by twp88** [comment]
@michaelherold Thank you for your response. 
When you say that it doesn't affect the functionality of the gem, do you refer to rubocop or haml-lint? Perhaps mine is a different error, but I seem to get no linting for haml files.

**Comment by michaelherold** [comment]
@twp88 The gem is working well for me. Here's an example:

```
~
↪ haml-lint blergh.html.haml
warning: parser/current is loading parser/ruby24, which recognizes
warning: 2.4.0-compliant syntax, but you are running 2.4.1.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
blergh.html.haml:1 [W] ClassesBeforeIds: Classes should be listed before IDs (.will-error should precede #this)
1 file inspected, 1 lint detected
~
↪ haml-lint --version
warning: parser/current is loading parser/ruby24, which recognizes
warning: 2.4.0-compliant syntax, but you are running 2.4.1.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
haml-lint 0.26.0
```

**Comment by lsarni** [comment]
I got the same error with `2.4.1`:
```
warning: parser/current is loading parser/ruby24, which recognizes
warning: 2.4.0-compliant syntax, but you are running 2.4.2.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
    at ChildProcess.<anonymous> (C:\Users\lulas\.atom\packages\linter-ruby\node_modules\sb-exec\lib\index.js:56:20)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:885:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
```
I'm using `linter-haml` on Atom and I get an error message on the corner every time.

**Comment by twp88** [comment]
@Isarni Does it actually do any linting, or does it just give you the error message? This issue appears to be for those who are just experiencing the error message, but with some files being linted in the process. In my case I get no linting and the error message.

**Comment by lsarni** [comment]
@twp88 the linting works fine in my case

**Comment by Superpencil** [comment]
Update, these https://github.com/whitequark/parser/issues/359 https://github.com/whitequark/parser/issues/336 bugs will be resolved soon and we'll have a new release

**Comment by joshRpowell** [comment]
@Superpencil the parser issues have been resolved.  Any update?

**Comment by Superpencil** [comment]
Doesn't look like it

**Comment by michaelherold** [comment]
Have you tried updating your version of `parser` with `bundle update parser` or, if you are not using Bundler, `gem update parser`? The version of Rubocop that we run -- which is where the `parser` dependency comes from -- [is constrained to allow whatever version above `2.3.3.1`](https://rubygems.org/gems/rubocop/versions/0.50.0).

You can also pin `parser` to a specific version if you like.

**Comment by sds** [comment]
Now that https://github.com/whitequark/parser/issues/359 https://github.com/whitequark/parser/issues/336 are closed, closing this issue.
